test/functional/forall/test-forall-atomic-utils.hpp

```diff
@@ -86,4 +88,73 @@ using AtomicDataTypeList =
               double
            >;
 
+
+// range segment multiplexer
+template< typename Index, typename SegType >
+struct RSMultiplexer {};
+
+template< typename Index >
+struct RSMultiplexer < Index, RAJA::TypedRangeSegment<Index> >
+{
+  RAJA::TypedRangeSegment<Index>
+  makeseg( Index N )
+  {
+    return RAJA::TypedRangeSegment<Index>( 0, N );
+  }
+};
+
+template< typename Index >
+struct RSMultiplexer < Index, RAJA::TypedRangeStrideSegment<Index> >
+{
+  RAJA::TypedRangeStrideSegment<Index>
+  makeseg( Index N )
+  {
+    return RAJA::TypedRangeStrideSegment<Index>( 0, N, 1 );
+  }
+};
+
+template< typename Index >
+struct RSMultiplexer < Index, RAJA::TypedListSegment<Index> >
+{
+  RAJA::TypedListSegment<Index>
+  makeseg( Index N )
+  {
+    std::vector<Index> temp(N);
+    std::iota( std::begin(temp), std::end(temp), 0 );
+    return RAJA::TypedListSegment<Index>( &temp[0], static_cast<size_t>(temp.size()) );
+  }
+};
+
+template< typename Index >
```

Comment:
@rchen20 you can remove the index set stuff. I think the segment stuff is sufficient.